tmol/optimization/sfxn_modules.py

```diff
@@ -14,20 +13,23 @@ def __init__(
         wpsm = score_function.render_whole_pose_scoring_module(pose_stack)
         self.whole_pose_scoring_module = wpsm
 
+        self.full_coords = pose_stack.coords
+        if coord_mask is None:
+            coord_mask = torch.full(
+                self.full_coords.shape[:-1],
+                True,
+                device=self.full_coords.device,
+                dtype=torch.bool,
+            )
         self.coord_mask = coord_mask
 
-        self.full_coords = pose_stack.coords
-        if self.coord_mask is None:
-            self.masked_coords = torch.nn.Parameter(pose_stack.coords)
-        else:
-            self.masked_coords = torch.nn.Parameter(pose_stack.coords[self.coord_mask])
+        self.masked_coords = torch.nn.Parameter(self.full_coords[self.coord_mask])
         self.count = 0
 
     def forward(self):
         self.count += 1
-        self.full_coords = DOFMaskingFunc.apply(
-            self.masked_coords, self.coord_mask, self.full_coords
-        )
+        self.full_coords = self.full_coords.detach()
+        self.full_coords[self.coord_mask] = self.masked_coords
```

Comment:
So was the performance issue related to this scatter but that `.detach()` above resolves it?

Reply:
there is a little overhead with the scatter (the 0.55s v 0.65s difference we currently see) but most of the overhead was from the memory allocation